src/source_clang.cc

```diff
@@ -919,6 +919,29 @@ Source::ClangViewAutocomplete(file_path, language) {
     return location;
   };
   
+  get_implementation_location=[this](const Token &token){
+    Offset location;
+    if(parsed && this->language && this->language->get_id()!="chdr" && this->language->get_id()!="cpphdr") {
+      for(auto &a_token: *clang_tokens) {
+        auto cursor=a_token.get_cursor();
+        auto kind=cursor.get_kind();
+        if((kind==clang::CursorKind::CXXMethod || kind==clang::CursorKind::Constructor || kind==clang::CursorKind::Destructor) &&
+           a_token.get_kind()==clang::Token_Identifier && cursor.has_type()) {
+          auto referenced=cursor.get_referenced();
+          if(referenced && static_cast<clang::CursorKind>(token.type)==referenced.get_kind() &&
+             token.spelling==a_token.get_spelling() && token.usr==referenced.get_usr()) {
+            location.file_path=cursor.get_source_location().get_path();
+            auto clang_offset=cursor.get_source_location().get_offset();
+            location.line=clang_offset.line;
+            location.index=clang_offset.index;
+            //do not break here, choose last occurrence in case declaration and implementation is in the same file
+          }
```

Comment:
Yeah, at some point later we should add message boxes, for instance top left of the window, that disappears after a few seconds. There might be support for this in gtk already, will investigate in the summer.